### PR TITLE
Migrate sample from GCM to FCM

### PR DESCRIPTION
**Pull request by lbalmaceda**
Followed this [migration guide](https://developers.google.com/cloud-messaging/android/android-migrate-fcm).
- Migrated `GcmListenerService` -> https://developers.google.com/cloud-messaging/android/android-migrate-gcmlistener 
- Removed unused `IDListenerService` -> https://developers.google.com/cloud-messaging/android/android-migrate-iid-service
- Updated the `app/README.md` file adding instructions on what values need to be set for this sample app to work. To sum up, the `google-services.json` file that can be downloaded from the Firebase console when opening the project configuration view, and the Guardian URL (same as before). 
- Additionally, ran the lint inspections and solved some of them.

**Sample app won't run unless a valid `google-services.json` file is present on the app's module root.**

### CODE REVIEW

File: app/src/main/java/com/auth0/guardian/sample/scanner/CaptureView.java [hunk 1]
```diff
@@ -123,17 +122,17 @@ private void createCameraSource(boolean autoFocus, boolean useFlash) {
                 .setFacing(CameraSource.CAMERA_FACING_BACK)
                 .setRequestedPreviewSize(1024, 600)
                 .setRequestedFps(15.0f)
-                .setFocusMode(autoFocus ? Camera.Parameters.FOCUS_MODE_CONTINUOUS_PICTURE : null);
+                .setFocusMode(true ? Camera.Parameters.FOCUS_MODE_CONTINUOUS_PICTURE : null);
```
Comment (santiagoaguiar) [inline app/src/main/java/com/auth0/guardian/sample/scanner/CaptureView.java]: We can remove the conditionals.

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -63,21 +63,23 @@
                     android:layout_width="match_parent"
                     android:layout_height="wrap_content"
                     android:layout_marginBottom="12dp"
+                    android:textIsSelectable="true"
```
Comment (santiagoaguiar) [inline app/src/main/res/layout/activity_main.xml]: Why this change?

File: guardian/src/main/java/com/auth0/android/guardian/sdk/GuardianAPIClient.java [hunk 3]
```diff
@@ -224,12 +224,12 @@ private String createJWT(@NonNull PrivateKey privateKey,
             String headerAndPayload = base64UrlSafeEncode(gson.toJson(headers).getBytes())
                     + "." + base64UrlSafeEncode(gson.toJson(claims).getBytes());
             final byte[] messageBytes = headerAndPayload.getBytes();
-            final Signature signer = Signature.getInstance("SHA256withRSA", "BC");
+            final Signature signer = Signature.getInstance("SHA256withRSA");
```
Comment (lbalmaceda) [inline guardian/src/main/java/com/auth0/android/guardian/sdk/GuardianAPIClient.java]:
```
Caused by: java.security.NoSuchAlgorithmException: The BC provider no longer provides an implementation for Signature.SHA256withRSA.  
```

Please see https://android-developers.googleblog.com/2018/03/cryptography-changes-in-android-p.html for more details.

File: app/src/main/java/com/auth0/guardian/sample/scanner/CaptureView.java (hunk 1 above)
Comment (lbalmaceda) [reply app/src/main/java/com/auth0/guardian/sample/scanner/CaptureView.java]: Agree. Left this to linter auto-fix.

File: app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (lbalmaceda) [reply app/src/main/res/layout/activity_main.xml]: **This only affects the sample app.** It makes the text selectable so one can easily copy the device name / fcm token and use it on the firebase console to send test fcm notifications . No cosmetic change.
